base/abstractarray.jl

```diff
@@ -422,6 +422,21 @@ function _maxlength(A, B, C...)
     max(length(A), _maxlength(B, C...))
 end
 
+# eachindex(A, d) and eachindex(A, (:, j))
+eachindex(A::AbstractArray, d::Integer) = 1:size(A, d)
```

Comment:
Isn't this ambiguous with the general `eachindex(A, B)` signature? You may end up with surprisingly different behavior for the same code given different inputs.